### PR TITLE
Pub 726 Add PIP Model

### PR DESCRIPTION
**Pull request by ChrisS1512**
PR not ready yet and marked as Draft.

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/PUB-726

### Change description ###

Initial data model for PIP.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

**Review by johnoneill46 — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/model/enums/Platform.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package uk.gov.hmcts.reform.pip.model.enums;
+
+/**
+ * Enum for platforms.
+ */
+public enum Platform {
+
+    CFT, CP
```
Comment (kurt-uk) [inline src/main/java/uk/gov/hmcts/reform/pip/model/enums/Platform.java:8]: ooo thast my bad, `platform` here is meant to refer to `hearing platform` so like skype or Teams etc

Comment (ChrisS1512) [reply src/main/java/uk/gov/hmcts/reform/pip/model/enums/Platform.java:8]: I've double checked on this one and I believe the above list to be the right ones for hearing platform however there may end up being more added, therefore I've I've left this as it is for now until the complete list is available

**Review by johnoneill46 — APPROVED** [review]

**Review by kurt-uk — APPROVED** [review]
👍🏼

File: src/main/java/uk/gov/hmcts/reform/pip/model/CourtRoom.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+public class CourtRoom {
+
+    @Id
+    private Long courtRoomId;
+
+    private Classification classification;
+
+    @Embedded
+    @AttributeOverrides({
+        @AttributeOverride(name = "value", column = @Column(name = "courtRoomName")),
```
Comment (kurt-uk) [inline src/main/java/uk/gov/hmcts/reform/pip/model/CourtRoom.java:29]: youve not used the constants in this file whereas you have in the others

**Review by kurt-uk — CHANGES_REQUESTED** [review]

File: src/main/java/uk/gov/hmcts/reform/pip/model/CourtRoom.java (hunk 2 above)
Comment (ChrisS1512) [reply src/main/java/uk/gov/hmcts/reform/pip/model/CourtRoom.java]: good spot :) updated

**Review by kurt-uk — APPROVED** [review]

**Pull request merged**